Codegen generates invalid java code (wrong types) for nested protobuf message in grpc call

I am using gradle codegen plugin latest version 1.1.0
my Protobuf looks like this:

``` 
syntax = "proto3";

option java_generate_equals_and_hash = true;
option java_package = "org.example.service.v1";

message EnvelopeTest {
  oneof type
  {
    RequestTest request = 4;
    ResponseTest response = 6;
  }
  message RequestTest {
  }
  message ResponseTest {
  }
}

service TestGrpcsService {
  rpc Transceive(EnvelopeTest.RequestTest) returns (stream EnvelopeTest.ResponseTest) {}
}
``````


However, the generated code does not compile due to wrong types, it's missing the EnvelopeTest in between,
e.g. manually changing <org.example.service.v1.Test.RequestTest> to <org.example.service.v1.Test.EnvelopeTest.RequestTest> solves the problem. 

e.g. 
.../build/generated/source/proto/main/pekkoGrpc/org/example/service/v1/TestGrpcsServiceClientPowerApi.java:19: error: cannot find symbol
      public StreamResponseRequestBuilder<org.example.service.v1.Test.RequestTest, org.example.service.v1.Test.ResponseTest> transceive()
                                                                     ^
  symbol:   class RequestTest
  location: class Test
```

--- a/codegen/src/main/scala/org/apache/pekko/grpc/gen/javadsl/Method.scala
+++ b/codegen/src/main/scala/org/apache/pekko/grpc/gen/javadsl/Method.scala
@@ -95,7 +95,10 @@ object Method {
     "_root_." + t.getFile.getOptions.getJavaPackage + "." + Service.protoName(t.getFile) + "." + t.getName
 
   /** Java API */
-  def getMessageType(t: Descriptor) = {
+  def getMessageType(t: Descriptor): String = {
+    if (t.getContainingType != null)
+      return getMessageType(t.getContainingType) + "." + t.getName
+
     val packageName =
       if (t.getFile.getOptions.hasJavaPackage) t.getFile.getOptions.getJavaPackage
       else t.getFile.getPackage

--- a/sbt-plugin/src/sbt-test/gen-java/06-nested-message/build.sbt
+++ b/sbt-plugin/src/sbt-test/gen-java/06-nested-message/build.sbt
@@ -1,0 +1,12 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * license agreements; and to You under the Apache License, version 2.0:
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * This file is part of the Apache Pekko project, which was derived from Akka.
+ */
+
+enablePlugins(PekkoGrpcPlugin)
+
+pekkoGrpcGeneratedLanguages := Seq(PekkoGrpc.Java)

--- a/sbt-plugin/src/sbt-test/gen-java/06-nested-message/project/plugins.sbt
+++ b/sbt-plugin/src/sbt-test/gen-java/06-nested-message/project/plugins.sbt
@@ -1,0 +1,10 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * license agreements; and to You under the Apache License, version 2.0:
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * This file is part of the Apache Pekko project, which was derived from Akka.
+ */
+
+addSbtPlugin("org.apache.pekko" % "pekko-grpc-sbt-plugin" % sys.props("project.version"))

--- a/sbt-plugin/src/sbt-test/gen-java/06-nested-message/src/main/java/org/example/service/v1/impl/TestGrpcsServiceImpl.java
+++ b/sbt-plugin/src/sbt-test/gen-java/06-nested-message/src/main/java/org/example/service/v1/impl/TestGrpcsServiceImpl.java
@@ -1,0 +1,23 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * license agreements; and to You under the Apache License, version 2.0:
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * This file is part of the Apache Pekko project, which was derived from Akka.
+ */
+
+package org.example.service.v1.impl;
+
+import org.apache.pekko.NotUsed;
+import org.apache.pekko.stream.javadsl.Source;
+
+import org.example.service.v1.TestGrpcsService;
+import org.example.service.v1.Test.EnvelopeTest.RequestTest;
+import org.example.service.v1.Test.EnvelopeTest.ResponseTest;
+
+class TestGrpcsServiceImpl implements TestGrpcsService {
+  public Source<ResponseTest, NotUsed> transceive(RequestTest request) {
+    throw new UnsupportedOperationException();
+  }
+}

--- a/sbt-plugin/src/sbt-test/gen-java/06-nested-message/src/main/protobuf/test.proto
+++ b/sbt-plugin/src/sbt-test/gen-java/06-nested-message/src/main/protobuf/test.proto
@@ -1,0 +1,20 @@
+syntax = "proto3";
+
+option java_generate_equals_and_hash = true;
+option java_package = "org.example.service.v1";
+
+message EnvelopeTest {
+  oneof type
+  {
+    RequestTest request = 4;
+    ResponseTest response = 6;
+  }
+  message RequestTest {
+  }
+  message ResponseTest {
+  }
+}
+
+service TestGrpcsService {
+  rpc Transceive(EnvelopeTest.RequestTest) returns (stream EnvelopeTest.ResponseTest) {}
+}

--- a/sbt-plugin/src/sbt-test/gen-java/06-nested-message/test
+++ b/sbt-plugin/src/sbt-test/gen-java/06-nested-message/test
@@ -1,0 +1,5 @@
+> compile
+
+$ exists target/scala-2.12/pekko-grpc
+$ exists target/scala-2.12/pekko-grpc/main/org/example/service/v1/TestGrpcsServiceClient.java
+$ exists target/scala-2.12/classes/org/example/service/v1/impl/TestGrpcsServiceImpl.class

